v2/emailpassword/nextjs/app-directory/init.mdx

```diff
@@ -298,10 +298,29 @@ export default function RootLayout({
 
 <CustomUIContent>
 
-## ^{nextjsinitlastnumber}) Call the frontend `init` functions
+## ^{nextjsinitlastnumber}) Call the frontend `init` functions and wrap with `<SuperTokensWrapper>` component
+
+- Create a client component `/app/components/supertokensProvider.tsx`. This file will initialise SuperTokens.
+- Modify the `/app/layout.tsx` file to use the `SuperTokensProvider` component. You can learn more about this file [here](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required).
+
+```tsx title="/app/components/supertokensProvider.tsx"
+'use client';
+import React from 'react';
+import SuperTokensWebJs from 'supertokens-web-js'
+// @ts-ignore
+import { frontendConfig } from '../config/frontend';
 
-- Modify the `/app/layout.tsx` file to initialise the SuperTokens SDK. You can learn more about this file [here](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required).
+if (typeof window !== 'undefined') {
+  // we only want to call this init function on the frontend, so we check typeof window !== 'undefined'
+  SuperTokensWebJs.init(frontendConfig());
+}
 
+export const SuperTokensProvider: React.FC<React.PropsWithChildren<{}>> = ({
+  children,
+}) => {
+  return <>{children}</>;
```

Comment:
i think you can just do `return children`?

Reply:
Yeah.. but getting some weird TS errors from the Code check plugin. Tried fixing it but I don't think it is worth it so left it as is.